kotlin/samples/dungeon/common/src/main/java/com/squareup/sample/dungeon/PlayerWorkflow.kt

```diff
@@ -57,15 +67,12 @@ class PlayerWorkflow(
     state: Movement,
     context: RenderContext<Movement, Nothing>
   ): Rendering {
+    val sink = context.makeSink<Action>()
+
     return Rendering(
         actorRendering = ActorRendering(avatar = avatar, movement = state),
-        onEvent = context.onEvent { event ->
-          val newMovement = when (event) {
-            is StartMoving -> state + event.direction
-            is StopMoving -> state - event.direction
-          }
-          enterState(newMovement)
-        })
+        onEvent = sink
```

Comment:
So I thought the swift style here was more like:

```
onEvent = { sink.send(it) }
```

Which keeps the screen types as pure functions. Don't know why they don't do this though. 

@apgar @benconchran?

Reply:
Not done yet, day is very fragmented. I'll make noise when the dust settles.